### PR TITLE
Refactoring Overall files

### PR DESCRIPTION
**Pull request by rubinstory**
Refactored Overall project files from Pusan National University.

### CODE REVIEW

**Comment by Grimler91** [comment]
Hi, 

* 218 commits and +2,599 -1,871 LOCs are pretty hard to review. Please split it up into smaller PRs (alternatively summarize the settings in the PR message, but that seems hard for such big changes as here). I see that you've already merged them in parts in [your repo](https://github.com/rubinstory/termux-app/pull/65), such PRs would be easier to review. 
* Out of the 218 commits in the PR there's 89 github merge commits that do not contribute anything
* The first two commits (three with the github merge commit) are [`test: git test with change readme`](https://github.com/termux/termux-app/pull/2786/commits/7adb5b92f2bc7b1d8c569e0d200030997b569990) and [`test`](https://github.com/termux/termux-app/pull/2786/commits/5bd947737645a1a530a65c1a5e568489ed63904b). Seems like those do not belong here

**Comment by rubinstory** [comment]
Even though we are just students learning how to refactor and apply, thank you for giving us your opinion despite our shortcomings. We will revise and reflect the commit based on your comments as soon as possible.

**Comment by Anonymous1231-dev** [comment]
@rubinstory  Hello , may I ask what this PR do?

**Comment by Anonymous1231-dev** [comment]
@rubinstory  i think you can squash all commit into one as it is too much commit (200+)

**Comment by agnostic-apollo** [comment]
Some of these commits do have very good improvements to code and can be merged, but some are not providing any meaningful benefit or seem to be just for the "sake of change". I'll try making a list later for what changes are not required and likely won't be approved. Obviously, these are too many changes and will take lot of time to review. Squashing some commits would be good, and as suggested, the merge commits should be removed.

Are there any specific requirements from your university about this or something we can help with?

Apart from refactoring, you could also look into existing issues and try to solve those.

Thanks.

**Comment by Anonymous1231-dev** [comment]
> Some of these commits do have very good improvements to code and can be merged, but some are not providing any meaningful benefit or seem to be just for the "sake of change". I'll try making a list later for what changes are not required and likely won't be approved. Obviously, these are too many changes and will take lot of time to review. Squashing some commits would be good, and as suggested, the merge commits should be removed.
> 
> Are there any specific requirements from your university about this or something we can help with?
> 
> Apart from refactoring, you could also look into existing issues and try to solve those.
> 
> Thanks.

Oh yeah, especially my issue #2769. It seems like be a big problem for me.

**Comment by rubinstory** [comment]
> 

Since there are too much commits as you said, our team is selecting commits that are "commit worthy". If you can give just a few examples of "good commit" from our request, it would be very helpful for us to rebase our commits.

Once more, thank you for giving advice to open source contributing newbies.

File: app/src/main/java/com/termux/app/TermuxService.java [hunk 1]
```diff
@@ -618,20 +658,23 @@ public synchronized int removeTermuxSession(TerminalSession sessionToRemove) {
     /** Callback received when a {@link TermuxSession} finishes. */
     @Override
     public void onTermuxSessionExited(final TermuxSession termuxSession) {
-        if (termuxSession != null) {
+        boolean isTermuxSession = termuxSession != null;
```
Comment (agnostic-apollo) [inline app/src/main/java/com/termux/app/TermuxService.java:661]: These and all the other if conditions extracted to boolean are useless and server no purpose, using an if condition already implies that one or more conditions will be supplied in it. Do not do this. If there are multiple conditions that are too complex, they can be split on multiple lines, in some case extraction to boolean is indeed fine.

**Comment by agnostic-apollo** [comment]
Do not modify the `SettingsActivity`, `RootPreferencesFragment` and the other `PreferenceDataStore`, or move them to individual class file. I kept  the fragments and datastores in same classes since they served similar functionalities and splitting would just create too many files without any need.

File: termux-shared/src/main/java/com/termux/shared/logger/Logger.java [hunk 2]
```diff
@@ -54,16 +54,33 @@ public class Logger {
...
-            Log.e(getFullTag(tag), message);
-        else if (logPriority == Log.WARN && CURRENT_LOG_LEVEL >= LOG_LEVEL_NORMAL)
-            Log.w(getFullTag(tag), message);
-        else if (logPriority == Log.INFO && CURRENT_LOG_LEVEL >= LOG_LEVEL_NORMAL)
-            Log.i(getFullTag(tag), message);
-        else if (logPriority == Log.DEBUG && CURRENT_LOG_LEVEL >= LOG_LEVEL_DEBUG)
-            Log.d(getFullTag(tag), message);
-        else if (logPriority == Log.VERBOSE && CURRENT_LOG_LEVEL >= LOG_LEVEL_VERBOSE)
-            Log.v(getFullTag(tag), message);
+        boolean higherThanNormal = (CURRENT_LOG_LEVEL >= LOG_LEVEL_NORMAL);
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/logger/Logger.java:57]: Don't do this. Need to minimize if conditions calculated for log statements.

File: termux-shared/src/main/java/com/termux/shared/settings/preferences/SharedPreferenceUtils.java [hunk 3]
```diff
@@ -52,7 +56,7 @@ public static SharedPreferences getPrivateAndMultiProcessSharedPreferences(Conte
      * default if failed to read a valid value, like in case of an exception.
      */
     public static boolean getBoolean(SharedPreferences sharedPreferences, String key, boolean def) {
-        if (sharedPreferences == null) {
+        if (isSharedPreferenceNull(sharedPreferences)) {
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/settings/preferences/SharedPreferenceUtils.java:59]: Don't do this.

File: termux-shared/src/main/java/com/termux/shared/termux/TermuxConstants.java [hunk 4]
```diff
@@ -668,6 +668,29 @@ public final class TermuxConstants {
     /** Termux and plugin apps directory */
     public static final File TERMUX_APPS_DIR = new File(TERMUX_APPS_DIR_PATH);
 
+    public static StringBuilder buildScript(String filesDir) {
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/TermuxConstants.java:671]: Don't do this. No the place for scripts.

File: terminal-view/src/main/java/com/termux/view/TerminalView.java [hunk 5]
```diff
@@ -94,7 +101,7 @@ public TerminalView(Context context, AttributeSet attributes) { // NO_UCD (unuse
             @Override
             public boolean onUp(MotionEvent event) {
                 mScrollRemainder = 0.0f;
-                if (mEmulator != null && mEmulator.isMouseTrackingActive() && !event.isFromSource(InputDevice.SOURCE_MOUSE) && !isSelectingText() && !scrolledWithFinger) {
+                if (!isEmulatorNull() && mEmulator.isMouseTrackingActive() && !event.isFromSource(InputDevice.SOURCE_MOUSE) && !isSelectingText() && !scrolledWithFinger) {
```
Comment (agnostic-apollo) [inline terminal-view/src/main/java/com/termux/view/TerminalView.java:104]: Don't do this.

File: termux-shared/src/main/java/com/termux/shared/android/AndroidUtils.java [hunk 6]
```diff
@@ -198,7 +216,11 @@ public static String getSystemPropertyWithAndroidAPI(@NonNull String property) {
 
     public static void appendPropertyToMarkdownIfSet(StringBuilder markdownString, String label, Object value) {
         if (value == null) return;
-        if (value instanceof String && (((String) value).isEmpty()) || "REL".equals(value)) return;
+        boolean isString = value instanceof String;
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/android/AndroidUtils.java:219]: Don't do this.

File: termux-shared/src/main/java/com/termux/shared/android/PermissionUtils.java [hunk 7]
```diff
@@ -73,7 +73,9 @@ public static boolean checkPermissions(@NonNull Context context, @NonNull String
         int result;
         for (String permission : permissions) {
             result = ContextCompat.checkSelfPermission(context, permission);
-            if (result != PackageManager.PERMISSION_GRANTED) {
+            boolean permissionGranted = (result == PackageManager.PERMISSION_GRANTED);
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/android/PermissionUtils.java:76]: Don't do this and below ones.

File: termux-shared/src/main/java/com/termux/shared/file/filesystem/FileAttributes.java [hunk 8]
```diff
@@ -265,25 +264,35 @@ public Set<FilePermission> permissions() {
         int bits = (st_mode & UnixConstants.S_IAMB);
         HashSet<FilePermission> perms = new HashSet<>();
 
-        if ((bits & UnixConstants.S_IRUSR) > 0)
+        boolean canOwnerRead = ((bits & UnixConstants.S_IRUSR) > 0);
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/file/filesystem/FileAttributes.java:267]: Don't do this. Can add comments if you must, although code pretty clear and is from AOSP.

File: termux-shared/src/main/java/com/termux/shared/reflection/ReflectionUtils.java [hunk 9]
```diff
@@ -158,16 +144,10 @@ public static boolean invokeVoidMethod(@NonNull Method method, Object obj, Objec
...
     }
 
-
-
     /** Class that represents result of invoking a method that has a non-void return type. */
-    public static class MethodInvokeResult {
-        public boolean success;
-        public Object value;
-
+    public static class MethodInvokeResult extends InvokeResult{
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/reflection/ReflectionUtils.java:148]: There must be a space before `{`.

File: termux-shared/src/main/java/com/termux/shared/shell/ShellUtils.java [hunk 10]
```diff
@@ -10,6 +10,7 @@
 public class ShellUtils {
 
     public static int getPid(Process p) {
+        int invalid = -1;
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/shell/ShellUtils.java:13]: Don't do this.

File: termux-shared/src/main/java/com/termux/shared/termux/crash/TermuxCrashUtils.java [hunk 11]
```diff
@@ -101,7 +101,9 @@ public void onPostLogCrash(final Context currentPackageContext, Thread thread, T
         }
 
         // If an uncaught exception, then do not notify since the termux app itself would be crashing
-        if (TYPE.UNCAUGHT_EXCEPTION.equals(mType) && TermuxConstants.TERMUX_PACKAGE_NAME.equals(currentPackageName))
+        boolean isUncaughtException = TYPE.UNCAUGHT_EXCEPTION.equals(mType);
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/crash/TermuxCrashUtils.java:104]: Don't do this.

File: termux-shared/src/main/java/com/termux/shared/termux/crash/TermuxCrashUtils.java [hunk 12]
```diff
@@ -199,27 +201,6 @@ private static synchronized void notifyAppCrashFromCrashLogFileInner(final Conte
         sendCrashReportNotification(context, logTag, null, null, reportString, false, false, null, false);
     }
 
-
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/crash/TermuxCrashUtils.java:202]: Don't remove these functions. These are library utility functions for callers.

File: termux-shared/src/main/java/com/termux/shared/termux/file/TermuxFileUtils.java [hunk 13]
```diff
@@ -26,6 +26,10 @@ public class TermuxFileUtils {
 
     private static final String LOG_TAG = "TermuxFileUtils";
 
+    private static boolean isStringHasValue(String str) {
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/file/TermuxFileUtils.java:29]: Skip this file.

File: termux-shared/src/main/java/com/termux/shared/termux/interact/TextInputDialogUtils.java [hunk 14]
```diff
@@ -12,6 +12,9 @@
 
 public final class TextInputDialogUtils {
 
+    public static final int PADDING_PER_DIP_AT_TOPANDSIDES = 16;
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/interact/TextInputDialogUtils.java:15]: Skip this file.

File: termux-shared/src/main/java/com/termux/shared/termux/notification/TermuxNotificationUtils.java [hunk 15]
```diff
@@ -25,11 +25,21 @@ public synchronized static int getNextNotificationId(final Context context) {
         int lastNotificationId = preferences.getLastNotificationId();
 
         int nextNotificationId = lastNotificationId + 1;
-        while(nextNotificationId == TermuxConstants.TERMUX_APP_NOTIFICATION_ID || nextNotificationId == TermuxConstants.TERMUX_RUN_COMMAND_NOTIFICATION_ID) {
+
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/notification/TermuxNotificationUtils.java:28]: Skip this file.

File: termux-shared/src/main/java/com/termux/shared/termux/plugins/TermuxPluginUtils.java [hunk 16]
```diff
@@ -207,27 +207,6 @@ public static void setPluginResultDirectoryVariables(ExecutionCommand executionC
             resultConfig.resultFileBasename = ShellUtils.getExecutableBasename(executionCommand.executable) + "-" + AndroidUtils.getCurrentMilliSecondLocalTimeStamp() + ".log";
     }
 
-
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/plugins/TermuxPluginUtils.java:210]: Skip this file.

File: termux-shared/src/main/java/com/termux/shared/termux/shell/am/TermuxAmSocketServer.java [hunk 17]
```diff
@@ -153,9 +154,13 @@ public static synchronized LocalSocketManager getTermuxAmSocketServer() {
     public static synchronized void showErrorNotification(@NonNull Context context, @NonNull Error error,
                                                           @NonNull LocalSocketRunConfig localSocketRunConfig,
                                                           @Nullable LocalClientSocket clientSocket) {
-        TermuxPluginUtils.sendPluginCommandErrorNotification(context, LOG_TAG,
-            localSocketRunConfig.getTitle() + " Socket Server Error", error.getMinimalErrorString(),
-            LocalSocketManager.getErrorMarkdownString(error, localSocketRunConfig, clientSocket));
+
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/shell/am/TermuxAmSocketServer.java:157]: Skip this file.

File: termux-shared/src/main/java/com/termux/shared/theme/ThemeUtils.java [hunk 18]
```diff
@@ -27,43 +27,37 @@ public static boolean isNightModeEnabled(Context context) {
     /** Will return true if mode is set to {@link NightMode#TRUE}, otherwise will return true if
      * mode is set to {@link NightMode#SYSTEM} and night mode is enabled by system. */
     public static boolean shouldEnableDarkTheme(Context context, String name) {
-        if (NightMode.TRUE.getName().equals(name))
+        boolean isNightMode = (NightMode.TRUE.getName().equals(name));
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/theme/ThemeUtils.java:30]: Skip this file.

File: termux-shared/src/main/java/com/termux/shared/theme/ThemeUtils.java [hunk 19]
```diff
@@ -27,43 +27,37 @@ public static boolean isNightModeEnabled(Context context) {
...
+        else {
             return false;
         }
     }
 
-
     /** Get {@link #ATTR_TEXT_COLOR_PRIMARY} value being used by current theme. */
     public static int getTextColorPrimary(Context context) {
-        return getSystemAttrColor(context, ATTR_TEXT_COLOR_PRIMARY);
+        return getSystemAttrColor(context, ATTR_TEXT_COLOR_PRIMARY, 0);
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/theme/ThemeUtils.java:45]: Intermediate library utility functions are often used that pass specific/default values.

File: termux-shared/src/main/java/com/termux/shared/view/ViewUtils.java [hunk 20]
```diff
@@ -110,23 +110,24 @@ public static Rect[] getWindowAndViewRects(View view, int statusBarHeight) {
                 ", displayOrientation=" + displayOrientation);
         }
 
+        boolean isStatusBarHeightTop = (statusBarHeight == windowRect.top);
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/view/ViewUtils.java:113]: Skip this file.

File: termux-shared/src/main/java/com/termux/shared/view/ViewUtils.java [hunk 21]
```diff
@@ -152,9 +153,14 @@ public static Rect[] getWindowAndViewRects(View view, int statusBarHeight) {
      */
     public static boolean isRectAbove(@NonNull Rect r1, @NonNull Rect r2) {
         // check for empty first
-        return r1.left < r1.right && r1.top < r1.bottom
-            // now check if above
-            && r1.left <= r2.left && r1.bottom >= r2.bottom;
+
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/view/ViewUtils.java:156]: Skip this too.

File: termux-shared/src/main/java/com/termux/shared/termux/extrakeys/ExtraKeysInfo.java [hunk 22]
```diff
@@ -151,18 +151,18 @@ private ExtraKeyButton[][] initExtraKeysInfo(@NonNull String propertiesInfo,
             for (int j = 0; j < matrix[i].length; j++) {
                 Object key = matrix[i][j];
 
-                JSONObject jobject = normalizeKeyConfig(key);
-
+                JSONObject jsonObject = normalizeKeyConfig(key);
```
Comment (agnostic-apollo) [inline termux-shared/src/main/java/com/termux/shared/termux/extrakeys/ExtraKeysInfo.java:154]: Skip this file too. Rename might be fine, not the if to boolean.

**Pull request closed without merging**